ocaml/parsing/location.mli

```diff
@@ -52,6 +52,12 @@ type t = Warnings.loc = {
    Else all fields are correct.
 *)
 
+(** Strict equality: Two locations are equal iff every field is equal.  Two
+    locations that happen to refer to the same place -- for instance, if one has
+    [pos_lnum] set correctly and the other has [pos_lnum = -1] -- are not
+    considered to be equal. *)
```

Comment:
I don't think there's anything surprising about the implementation or semantics of `equal` here, and I'd vote to remove the comment as such.

Reply:
I think I'd like somebody who thinks about locations more regularly to weigh in on this -- do users think there's an abstraction boundary around `Location.t` or not?

Reply:
Leaving the comment seems fine. This is just a minor comment of mine. It's fine with me if you want to resolve this conversation.